helper/types.js

```diff
@@ -0,0 +1,41 @@
+var valid_types = require( '../query/types' );
```

Comment:
splitting this out into a separate file is awesome. I wonder if the filename should be a bit more descriptive, since `type` is such an overloaded term.